src/sentry/notifications/notificationcontroller.py

```diff
@@ -58,14 +61,24 @@ def __init__(
         organization_id: int | None = None,
         type: NotificationSettingEnum | None = None,
         provider: ExternalProviderEnum | None = None,
+        organization: Organization | None = None,
     ) -> None:
-        self.recipients = recipients
+        if features.has("organizations:team-workflow-notifications", organization):
+            self.recipients = []
```

Comment:
mypy issue rn: `error: Incompatible types in assignment (expression has type "Union[Iterable[RpcActor], Iterable[Team], Iterable[RpcUser]]", variable has type "List[Union[RpcActor, Team, RpcUser]]")`
- changing this line so that i set the type to the expected type causes issues with the recipient processing added below (ie. `len`, `append` don't match with `Iterable`)
- changing the type of `recipients` in the signature causes issues in files that instantiate a `NotificationController` (ie. "expected `List[Union[RpcActor, Team, RpcUser]]` but got `List[RpcActor]`")